### PR TITLE
Add test with jest

### LINKED ISSUE #18: Add test
Test with jest on API side and front side with react testing library

### PR DESCRIPTION
**Pull request by zyhou**
resolve #18 

- [x] add jest with react testing library
- [x] minimal test on two function and one component
- [x] ignore file on now.sh (alredy https://zeit.co/docs/v2/build-step/#ignored-files-and-folders)

image

### CODE REVIEW

File: .babelrc [hunk 1]
```diff
@@ -1,12 +1,27 @@
...
-    [
-      "import",
-      {
-        "libraryName": "antd",
-        "style": "css"
-      }
-    ]
-  ]
+    "presets": ["next/babel"],
+    "env": {
```
Comment (zyhou) [inline .babelrc:3]: Need to do this because 'import' plugin bug in test env ie https://laysent.github.io/til/2019-07-29_jest-with-ant-design

**Comment by Euregan** [comment]
#### 📝Changed routes:


| [**/talks**](https://aperoweb-git-add-test-with-jest.euregan.now.sh/talks) |
|:-:|
|<a href="https://aperoweb-git-add-test-with-jest.euregan.now.sh/talks">Screenshot of /talks</a><br /><sup><a href="https://images.weserv.nl?url=microlink.nyc3.digitaloceanspaces.com/kEnJ134IQkjlTYTBPoBYojp22n4_">(view full size)</a> |



And 4 other routes:
- [**/api/calendar**](https://aperoweb-git-add-test-with-jest.euregan.now.sh/api/calendar)
- [**/api/calendar.spec**](https://aperoweb-git-add-test-with-jest.euregan.now.sh/api/calendar.spec)
- [**/api/talks**](https://aperoweb-git-add-test-with-jest.euregan.now.sh/api/talks)
- [**/api/talks.spec**](https://aperoweb-git-add-test-with-jest.euregan.now.sh/api/talks.spec)



Commit 5dc2dc0f2f8c4b74476ad10577efc8d487f78253 (https://aperoweb-git-add-test-with-jest.euregan.now.sh).

**Comment by zyhou** [comment]
> * [**/api/calendar**](https://aperoweb-git-add-test-with-jest.euregan.now.sh/api/calendar)
> * [**/api/calendar.spec**](https://aperoweb-git-add-test-with-jest.euregan.now.sh/api/calendar.spec)
> * [**/api/talks**](https://aperoweb-git-add-test-with-jest.euregan.now.sh/api/talks)
> * [**/api/talks.spec**](https://aperoweb-git-add-test-with-jest.euregan.now.sh/api/talks.spec)

Oh now.sh push .spec files, need to exclude this files.

Can you give me access to the now.sh project ? I can't access to https://aperoweb-git-add-test-with-jest.euregan.now.sh/_src

File: pages/api/talks.spec.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+    describe('getNextEmptyMonth', () => {
+        it.each`
+            talks                                                                       | expected
+            ${[{ date: '2019-11-20' }]}                                                 | ${1}
+            ${[{ date: '2019-12-20' }, { date: '2020-01-20' }]}                         | ${2}
+            ${[{ date: '2019-12-20' }, { date: '2020-01-20' }, { date: '2020-02-20' }]} | ${3}
+            ${[{ date: '2019-12-20' }, { date: '2020-02-20' }]}                         | ${1}
+            ${[{ date: '2020-02-20' }]}                                                 | ${1}
+        `('should return the right month: $expected', ({ talks, expected }) => {
```
Comment (Euregan) [inline pages/api/talks.spec.js:19]: What the hell is that? :D

**Comment by Euregan** [comment]
> Can you give me access to the now.sh project ? I can't access to https://aperoweb-git-add-test-with-jest.euregan.now.sh/_src

I can't, multiple users are on a paid plan :(

File: pages/api/talks.spec.js (hunk 2 above)
Comment (zyhou) [reply pages/api/talks.spec.js:19]: [it.each doc](https://jestjs.io/docs/en/api#describeeachtablename-fn-timeout)

Really powerfull function. When you write same tests but only params change you can use it.each. 
Jest expose two style :
- Tagged template literal, like above
- Array like 

```
it.each([
  [1, 1, 2],
  [1, 2, 3],
  [2, 1, 3],
])('.add(%i, %i)', (a, b, expected) => {
    expect(a + b).toBe(expected);
  });
```

Comment (Euregan) [reply pages/api/talks.spec.js:19]: Not a big fan of the template as a data structure thing, but okay

**Review by Euregan — APPROVED** [review]

**Pull request merged**